scripts/data/gen-nodetime/src/relayer/lib/xrelayer.ts

```diff
@@ -0,0 +1,490 @@
+			? this.config.paths.find((x) => x.path.id == pathID)
+			: null;
+	}
+	protected async balanceCheck(chain: ChainConfig): Promise<boolean> {
+		let chainBalances = await this.getDefaultAccountBalance([chain.chainId]);
+		let chainGP = GasPrice.fromString(chain.gasPrice);
+		if (!chainBalances.find((x) => x.denom == chainGP.denom)) {
+			return false;
+		}
+		if (
+			chainBalances.find(
+				(x) =>
+					x.denom == chainGP.denom &&
+					parseInt(x.amount) < chainGP.amount.toFloatApproximation() * 2256000
+			)
+		) {
+			return false;
+		}
+		return true;
+	}
+	protected async createLink({ path, options }: PathConfig): Promise<void> {
+		let chainA = this.chainById(path.src.chainID);
+		let chainB = this.chainById(path.dst.chainID);
+		let chainAGP = GasPrice.fromString(chainA.gasPrice);
+		let chainBGP = GasPrice.fromString(chainA.gasPrice);
+		if (!(await this.balanceCheck(chainA))) {
+			throw new Error(
+				"Not enough balance available on '" +
+					chainA.chainId +
+					"'. You need at least " +
+					chainAGP.amount.toFloatApproximation() * 2256000 +
+					chainAGP.denom
+			);
+		}
+		if (!(await this.balanceCheck(chainB))) {
+			throw new Error(
+				"Not enough balance available on '" +
+					chainB.chainId +
+					"'. You need at least " +
+					chainBGP.amount.toFloatApproximation() * 2256000 +
+					chainBGP.denom
+			);
+		}
+		// Create IBC Client for chain A
+
+		const clientA = await this.getIBCClient(chainA);
+		// Create IBC Client for chain B
+
+		const clientB = await this.getIBCClient(chainB);
+
+		const link = await Link.createWithNewConnections(clientA, clientB);
+		const channels = await link.createChannel(
+			"A",
+			options.sourcePort,
+			options.targetPort,
+			orderFromJSON(options.ordering),
+			options.targetVersion
+		);
+		let configPath = this.pathById(path.id);
+		configPath.path.src.channelID = channels.src.channelId;
+		configPath.path.dst.channelID = channels.dest.channelId;
+		configPath.path.isLinked = true;
+		configPath.connections = {
+			srcConnection: link.endA.connectionID,
+			destConnection: link.endB.connectionID,
+		};
+		configPath.relayerData = null;
+	}
+	protected async getLink({ path, connections }: PathConfig): Promise<Link> {
+		let chainA = this.chainById(path.src.chainID);
+		let chainB = this.chainById(path.dst.chainID);
+		// Create IBC Client for chain A
+
+		const clientA = await this.getIBCClient(chainA);
+		// Create IBC Client for chain B
+
+		const clientB = await this.getIBCClient(chainB);
+
+		const logger = new ConsoleLogger();
+		const link = Link.createWithExistingConnections(
+			clientA,
+			clientB,
+			connections.srcConnection,
+			connections.destConnection,
+			logger
+		);
+		return link;
+	}
+	protected async getIBCClient(chain: ChainConfig): Promise<IbcClient> {
+		let chainGP = GasPrice.fromString(chain.gasPrice);
+		let signer = await DirectSecp256k1HdWallet.fromMnemonic(
+			this.config.mnemonic,
+			{
+				hdPaths: [stringToPath("m/44'/118'/0'/0/0")],
```

Comment:
needs to be defined as a constant

Reply:
actually I've left this for now because this needs to be a starport relayer configure param.... I was gonna ask you guys about this

Most 3rd party chains DON'T use this hdPath

Reply:
HDPaths to be configurable in future version